errors.go

```diff
@@ -0,0 +1,69 @@
+package yoti
+
+import (
+	"fmt"
+)
+
+// InvalidKeyError describes an error with a private key
+type InvalidKeyError struct {
+	nestedError error
+}
+
+func (err InvalidKeyError) Error() string {
+	return "Invalid Private Key"
+}
+
+// Unwrap obtains the wrapped error
+func (err InvalidKeyError) Unwrap() error {
+	return err.nestedError
+}
+
+// HTTPError describes an error returned from the remote HTTP/REST API
+type HTTPError struct {
```

Comment:
I would advise against having an `HTTPError` unless it is for internal use (in which case it should be private). Because it will require users of the SDK to understand what http error codes can be returned from services that the sdk interacts with, which puts more burden on them.

Also isn't it intended that the SDK abstracts the internals of the Yoti services away from it's users?

Reply:
So we discussed this offline:

I was given the impression that part of this work was that we had a customer who wanted a list of error values for them to handle (how I wasn't clear) in their business logic. We already provide a success/fail value where err != nil - with a string that can be used by support for debugging/logging but one that can't be programmatically compared for flow control (go compares errors by reference, not value).

Ed suggested having a meeting about what we want the requirements to be when everyone's in the office (Tuesday maybe?)

Reply:
Sounds good :+1: